### ISSUE TITLE
Chips: Removed in V18

### ISSUE THREAD

**Issue by PierreB40** [primefaces/primeng#16390]
As I said in this issue: https://github.com/primefaces/primeng/issues/16091
The deletion of the Chips component in V18 seems to me to be a mistake.
Autocomplete and chips don't have the same parameters, and the same uses.
The proposed alternative is not flexible enough: if chips disappears, autcomplete must be able to cover 100% of the possibilities offered by Chips.

**Comment by vetemi** [comment]
I second this! Not only from a technical point of view but also the way this happened. You simply don't remove functionality from your library without deprecating it first and give time to adapt...

**Comment by nico-belonio** [comment]
Is that a mistake? What alternatives are there?

**Comment by Yuri-Lima** [comment]
> I second this! Not only from a technical point of view but also the way this happened. You simply don't remove functionality from your library without deprecating it first and give time to adapt...

Exactly, with all due respect, but without deprecation is something that makes the library not reliable to use. It doesnt make any sense to remove something out of the blue, without having a smooth process code migration.

Please, guys do at least a deprecation.

**Comment by nchaigne** [comment]
Same. I'm using Chips, tried to upgrade from primeng 17 but can't since the proposed alternative "autocomplete" does not offer equivalent features.

**Comment by Henoch0** [comment]
Same here. I recently started a new project and chose PrimeNG, unaware that it lacks an equivalent feature for Chips, which is quite surprising and disappointing.

**Comment by m3allimm4a4** [comment]
Technically you can use the autocomplete component using the multiple plus the without tipeahead option
[https://primeng.org/autocomplete#multiple](url)

**Comment by jvelazquez29** [comment]
I would like to add that I too agree that it should either be deprecated or at least add the separator prop to autocomplete. My current use case of separator was to be able to copy and paste column of data from excel to the component and with a RegExp separate the values into individual chips instead of the whole data in a single chip.

**Comment by nchaigne** [comment]
I'm trying to use "p-autocomplete" instead of "p-chips", but I couldn't find an equivalent to "addOnBlur" that I need.
There is a "onBlur" callback, but it does not provide anything useful in the event (only {"isTrusted":true}).

Is it possible to get the current input value in "onBlur" ?

If not, would it be possible to implement an equivalent to "addOnBlur" from "p-chips" ?

**Comment by sujeetjaiswara** [comment]
This fix works equivalently to 'addOnBlur' 🙂

```html
<p-autocomplete         
    formControlName="tags"
    placeholder="Add tags"
    multiple fluid 
    [typeahead]="false"
    (onBlur)="onBlurMyAutocomplete($event)"
 />
 ```

```typescript
onBlurMyAutocomplete(e: Event) {
    const inputElement = e.target as HTMLInputElement;
    const inputValue = inputElement.value; 
    if(inputValue) {
      inputElement?.dispatchEvent(new KeyboardEvent('keydown', { 'code': 'Enter' } ));
    }    
}
```

**Comment by arthurb4a** [comment]
I used chips to paste content from a sheet, so it would separate th values into chips.

My workaround was to use a textarea subscribing to (input) and (blur) events.
Then I use a regex to test for the keys I want to use as separators and split the string.
Ex: ```/\r|\n|;|,| /```

Then I show the separated values as a p-chip elements on the bottom to show the "expected" behavior.
Worked for my case.

Still, PrimeNG should deprecate its elements before they are removed!

**Comment by adama357** [comment]
According to [PrimeNG's own documentation](https://primeng.org/autocomplete), "AutoComplete is an input component that provides real-time suggestions when being typed." That is completely different from the purpose of the chips component. 

And not only does AutoComplete not include many of the options provided by Chips, but the out-of-the-box functionality is different, even if you have multiple enabled and typeahead disabled. 

Example: So let's say I'm filling out a form, and one of the form fields is a chips input. Using the (removed) Chips component, if I typed in something in the field and then clicked off of it (moving on in the form), it would automatically add a chip for the text I entered. But if I use the AutoComplete component instead (with multiple enabled and typeahead disabled), if I just type in text and then click off of it, the text remains, but it doesn't create a chip. I need to go back to the input and hit enter. 

Right now, it's looking like the best option is to just manually add back in [PrimeNG 17's Chips component](https://github.com/primefaces/primeng/blob/v17/src/app/components/chips/chips.ts).

**Comment by mcrtricolor** [comment]
I also need chips to upgrade my project, i use it too much.

I also use autocomplete and don't understand how these two are compatible.

Thanks in advance!

**Comment by kristiyanpetrov** [comment]
do you going to add back just to v18 and not to v19 as well?

**Comment by cagataycivici** [comment]
v18 is in LTS stage so need to think about it but definitely will try for v19. Overall, as the founder of PrimeNG I was away from the project for the past 2 years to focus on other projects, however I'm now back as the project manager. My primary goal is starting with v20, every update should be drop-in replacement. More public information coming soon.

**Comment by kristiyanpetrov** [comment]
> v18 is in LTS stage so need to think about it but definitely will try for v19. Overall, as the founder of PrimeNG I was away from the project for the past 2 years to focus on other projects, however I'm now back as the project manager. My primary goal is starting with v20, every update should be drop-in replacement. More public information coming soon.

i see, the issue that i have in my current project is that we used v17 and the implementation i did with chips was great for my use case and now after we updated to v19 i can't think for any straightforward way to achieve same result, that's why i was hope chips might be returned in v19 as other guys mentioned that autocomplete can't replace chips :)

**Comment by cagataycivici** [comment]
What is the issue with non typeahead mode of AutoComplete, here is [demo](https://primeng.org/autocomplete#multiple). Plan was to deprecate Chips, and offer AutoComplete Multiple with typeahead false but the team made an error and delete Chips.

**Comment by kristiyanpetrov** [comment]
> What is the issue with non typeahead mode of AutoComplete, here is [demo](https://primeng.org/autocomplete#multiple). Plan was to deprecate Chips, and offer AutoComplete Multiple with typeahead false but the team made an error and delete Chips.

hmm okay it seems i've miss this but this might work for me, i will try it tomorrow and will report back
<img width="367" alt="Image" src="https://github.com/user-attachments/assets/4fe580ee-71e1-4e4e-af7b-86a154e0bc47" />

**Comment by kristiyanpetrov** [comment]
@cagataycivici i just tested the example but if i place bulk with numbers for ex. 231213213213 231232121123 2132131232131 213312321321 231321321321 21321321321321 21321321321321 3213213213213 2132132132121

they appear in one chip instead each number to be in separate chip (with p-chips this was worked as expected)

**Comment by stephanie-dm** [comment]
@cagataycivici : Will p-chips be added again to PrimeNG 19.X? This is a breaking change for us, and we can't wait until version 20 to have this added again. We just spent a lot of time upgrading all our applications to Angular + PrimeNG 19 and now we stumbled upon this almost at the end of the upgrade... 

We have the PRO support, so I wanted to create a ticket in your JIRA, but I can't log in anymore. When I try to click "contact your JIRA administrator", I get this message: "Your JIRA administrator has not yet configured this contact form.".

**Comment by Ronkiro** [comment]
Bump, i just got into this and it's kinda bad i will have to nearly implement it by hand

**Comment by cagataycivici** [comment]
InputChips is now re-added and published in v18.0.2-patch.1, next week also for v19.1.0.

**Comment by mcrtricolor** [comment]
> InputChips is now re-added and published in v18.0.2-patch.1, next week also for v19.1.0.

Could it be reviewed the deprecation of this? We need Chips, it's not the same as autocomplete.

Thanks for the fix!

**Comment by adama357** [comment]
Thank you @cagataycivici ! 

> We need Chips, it's not the same as autocomplete.

Right, if the plan is definitely to combine Chips into AutoComplete, then I think AutoComplete needs to be renamed. 

If you add all of Chips' functionality to AutoComplete and turn on the correct settings to make it behave the same as Chips, at that point calling it "auto-complete" is very confusing.

**Comment by cagataycivici** [comment]
Yes, but our goal is to make PrimeNG as stable as possible now so renaming a popular component would be the opposite. Instead, we may improve documentation with Chips section and make sure it works exactly as InputChips.

**Comment by cagataycivici** [comment]
> > InputChips is now re-added and published in v18.0.2-patch.1, next week also for v19.1.0.
> 
> Could it be reviewed the deprecation of this? We need Chips, it's not the same as autocomplete.
> 
> Thanks for the fix!

We'll make sure it works exactly like InputChips, e.g. handling of spaces and such. No plans to remove it in near future.

**Comment by gregorAutomatics** [comment]
I'm currently migrating from v17 to v19. And so I am keeping tabs on what the changes are for v20.
The deprecated components are now fully removed. Including Chips.
Is this now final? Autocomplete still doesn't provide equivalent functionalites to replace it

**Comment by cs-StephanieDM** [comment]
@gregorAutomatics : I'm using PrimeNG 19.1.3 and I could use the p-chips component again, since they re-added it.

**Comment by Ronkiro** [comment]
Yes they've readded, see: https://github.com/primefaces/primeng/issues/16390#issuecomment-2795088855

> InputChips is now re-added and published in v18.0.2-patch.1, next week also for v19.1.0.

**Comment by gregorAutomatics** [comment]
@Ronkiro 
@stephanie-dm 
@cagataycivici 
It's not an issue for v19. It has been removed again in v20: https://github.com/primefaces/primeng/commit/86389f5126bd484727401c823fe00f780425fbfa

**Comment by mcrtricolor** [comment]
> InputChips is now re-added and published in v18.0.2-patch.1, next week also for v19.1.0.

Hi, I'm using v18.0.2-patch.1 and i cannot find ChipsModule anywhere.

How can I use it?

Thanks in advance!

**Comment by lblu12555** [comment]
It looks like chips is removed again in v20.  Right now autocomplete seems to be missing an onAdd type emitter when typeahead=false and multiple=true, (just an onSelect event when a typeahead suggestion is selected).  This is what this issue https://github.com/primefaces/primeng/issues/17114 was originally reporting.
